### PR TITLE
Replace style file with directives to not sort vulkan_headers.h

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
The internal formatter does not respect `.clang-format` files for...
reasons. This method works without having to use the style file. It's
also nice for humans to have the comment about what's going on. It's a
bit verbose, unfortunately.

### CODE REVIEW

**Review by benvanik — CHANGES_REQUESTED** [review]
not a fan of this one - why is this needed? clang-format should not reorder headers in different groups because of:
https://github.com/google/iree/blob/e44dee562e1d25bc26a1050db2bc7fac4e70bb4a/.clang-format#L20-L28

**Comment by benvanik** [comment]
(if we *really* need this for some internal reason, I think if you put a comment on the header line it won't get sorted - it'd be nicer to have `// Required to be included before all other headers:` or something that is useful to humans)

**Comment by GMNGeoffrey** [comment]
> not a fan of this one - why is this needed? (clang-format should not reorder headers in different groups)

Hmmm I was hitting this when trying to fix up our includes internally. Apparently our internal version does :upside_down_face: If this only happens with the internal version then I'll just revert the diffs I just got. Someone editing those files will have to be careful :confused:

**Comment by benvanik** [comment]
Is .clang-format not respected internally, or are we excluding it from copybara? I thought (awhile back) cider at least could use those files if present.

**Comment by GMNGeoffrey** [comment]
Ah I see it's specifically because of that directive in `.clang-format`. That explains why the internal version ignores it: it doesn't respect style files.

**Comment by GMNGeoffrey** [comment]
> (if we _really_ need this for some internal reason, I think if you put a comment on the header line it won't get sorted - it'd be nicer to have `// Required to be included before all other headers:` or something that is useful to humans)

Putting a comment on the header line doesn't prevent sorting. I searched around and `clang-format off/on` was the only way I found. I would also prefer something that could be inline and useful for humans.

**Comment by GMNGeoffrey** [comment]
`clang-format` will sort across groups (and create groups) based on `IncludeBlocksStyle`, which is set to Regroup in google style: https://github.com/llvm/llvm-project/blob/main/clang/lib/Format/Format.cpp#L1026

File: iree/hal/vulkan/internal_vk_mem_alloc.h [hunk 1]
```diff
@@ -16,7 +16,9 @@
 #define IREE_HAL_VULKAN_INTERNAL_VK_MEM_ALLOC_H_
 
 // NOTE: ensure our vulkan headers are used (as we define VK_NO_PROTOTYPES).
```
Comment (benvanik) [inline iree/hal/vulkan/internal_vk_mem_alloc.h:18]: dupe here

**Review by benvanik — APPROVED** [review]

File: iree/hal/vulkan/internal_vk_mem_alloc.h (hunk 1 above)
Comment (GMNGeoffrey) [reply iree/hal/vulkan/internal_vk_mem_alloc.h]: Done

**Comment by GMNGeoffrey** [comment]
Of course now is the time that our clang-format presubmit decides to break, but I tested it locally with  `git clang-format main --binary=clang-format-9 --style=file`

**Pull request merged**